course7/homeworks/patrickkostjens/ls/ls.cc

```diff
@@ -1,13 +1,60 @@
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 #include <iostream>
 
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+size_t get_size(fs::path path)
+{
+	if (!fs::is_directory(path)) return fs::file_size(path);
+
+	size_t size = 0;
+	for (const auto& it : fs::directory_iterator(path)) 
+		size += get_size(it);
+	return size;
+}
+
+void print_contents(fs::path path, bool extensive, bool tree, int level = 0)
+{
+	const auto& is_directory = fs::is_directory(path);
```

Comment:
```is_directory``` returns simple boolean, a 1-byte number. There is no point storing a reference to it.